src/ethereum/forks/amsterdam/vm/instructions/storage.py

```diff
@@ -91,35 +87,28 @@ def sstore(evm: Evm) -> None:
         The current EVM frame.
 
     """
+    # check static context before accessing storage
+    if evm.message.is_static:
+        raise WriteInStaticContext
+
     # STACK
     key = pop(evm.stack).to_be_bytes32()
     new_value = pop(evm.stack)
 
     # check we have at least the stipend gas
     check_gas(evm, GAS_CALL_STIPEND + Uint(1))
 
-    # check static context before accessing storage
-    if evm.message.is_static:
-        raise WriteInStaticContext
-
-    # GAS
-    gas_cost = Uint(0)
-    is_cold_access = (
-        evm.message.current_target,
-        key,
-    ) not in evm.accessed_storage_keys
-
-    if is_cold_access:
-        gas_cost += GAS_COLD_SLOAD
-
     state = evm.message.block_env.state
     original_value = get_storage_original(
         state, evm.message.current_target, key
     )
     current_value = get_storage(state, evm.message.current_target, key)
 
-    if is_cold_access:
+    gas_cost = Uint(0)
```

Comment:
This also returns the logic to what it is, no need to change this unless we want the `is_cold_access` check on all operations. I think we can decide this on the isolated PR as well. I'd rather not alter too much of the underlying code in our changes so we can more easily rebase up to eip-7928.